versioned/storage/cache/src/main/java/org/projectnessie/versioned/storage/cache/CaffeineCacheBackend.java

```diff
@@ -154,6 +167,52 @@ public void clear(@Nonnull String repositoryId) {
     cache.asMap().keySet().removeIf(k -> k.repositoryId.equals(repositoryId));
   }
 
+  @Override
+  public void removeReference(@Nonnull String repositoryId, @Nonnull String name) {
+    if (refCacheTtlNanos <= 0L) {
+      return;
+    }
+    ObjId id = objIdFromByteArray(name.getBytes(UTF_8));
```

Comment:
Could it clash with another `ObjId`? This seems risky to me as `name` is essentially user-provided. I think we should segregate reference keys from normal `ObjId` keys (e.g. add another field to `CacheKeyValue`)

Reply:
Good point! addressed